File: avance-salaire-backend/src/main/java/com/soprahr/avancesalairebackend/model/dto/UserDTO.java
package com.soprahr.avancesalairebackend.model.dto;

import com.soprahr.avancesalairebackend.model.enums.UserRole;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class UserDTO {
    private Long id;
    private String name;
    private String lastName;
    private String firstName;
    private String jobTitle;
    private String email;
    private String phone;
    private boolean status;
    private UserRole role;
    private String company;
    private BigDecimal salary;
    private String profilePicture;
    // Add other fields as needed, but NOT password!
}

File: avance-salaire-frontend/src/app/core/models/user.dto.ts
import { UserRole } from './user-role.enum';

export interface UserDTO {
  id: number;
  name: string;
  lastName: string;
  firstName: string;
  jobTitle: string;
  email: string;
  phone: string;
  status: boolean;
  role: UserRole;
  company: string;
  profilePicture?: string;
}
